commons/src/main/kotlin/org/fossify/commons/extensions/Activity.kt

```diff
@@ -226,30 +226,82 @@ fun BaseSimpleActivity.isShowingSAFCreateDocumentDialogSdk30(path: String): Bool
+                        positive = android.R.string.ok,
+                        negative = 0
+                    ) {}
+                } else {
+                    ConfirmationAdvancedDialog(
+                        this,
+                        "",
+                        R.string.confirm_storage_access_restricted_text,
+                        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.UPSIDE_DOWN_CAKE) {
+                            R.string.confirm_storage_access_restricted_text_open_system
+                        } else {
+                            R.string.confirm_storage_access_restricted_text_request_access
+                        },
+                        R.string.cancel
+                    ) { success ->
+                        if (success) {
+                            val uri = createAndroidDataOrObbUri(path)
+
+                            // On Android 14+, there is no longer any workaround to access the data/ and obb/ folders
+                            // Open the system app instead
+                            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.UPSIDE_DOWN_CAKE) {
+                                // Using a if with OR conditions allows to test them one by one, and stops at the first successful one
+                                if (
+                                    startIntentForUriAction(
+                                        uri,
+                                        "android.intent.action.VIEW",
+                                        ComponentName("com.google.android.documentsui", "com.android.documentsui.files.FilesActivity")
+                                    ) ||
+                                    startIntentForUriAction(
+                                        uri,
+                                        "android.intent.action.VIEW",
+                                        ComponentName("com.android.documentsui", "com.android.documentsui.files.FilesActivity")
+                                    ) ||
+                                    startIntentForUriAction(
+                                        uri,
+                                        "android.intent.action.VIEW",
+                                        ComponentName("com.android.documentsui", "com.android.documentsui.FilesActivity")
+                                    ) ||
+                                    startIntentForUriAction(uri, "android.intent.action.VIEW", null) ||
+                                    startIntentForUriAction(uri, "android.provider.action.BROWSE", null) ||
+                                    startIntentForUriAction(uri, "android.provider.action.BROWSE_DOCUMENT_ROOT", null)
+                                ) {
+                                    return@ConfirmationAdvancedDialog
+                                } else {
+                                    toast(R.string.confirm_storage_access_restricted_text_not_found)
+                                }
```

Comment:
If we must handle all these cases, it would be cleaner to extract this to an extension function. 

Maybe `launchSystemFileManager()`?